### DISCUSSION TITLE
Livewire Tabs

### DISCUSSION THREAD

**Discussion by mvpopuk** [livewire/livewire#2861] [1. Help] [answered]
Hello everyone, 

I know you are all busy caught up coding your own projects but I may need some help to better understand some principles.

**The problem**

So I have tabs dynamically generated... those tabs are actually items from a Model. Every time the user adds a new item it will be displayed as a tab.

My code at the moment is:

```
<?php

namespace App\Http\Livewire;

use Livewire\Component;
use App\Produse;
use App\Corpuri;

class CatalogCorpuri extends Component
{
    public $produs;
    public $corpuri;
    public bool $arataCorp = false;

    public function mount($id)
    {
        $produs = Produse::find($id);
        $prod_id = $produs->id;
        $this->corpuri = Corpuri::where('prod_id', $prod_id)->get();
    }

    public function render()
    {
        return view('livewire.catalog-corpuri');
    }
}
```
And in the view, I just loop thru `$corpuri` and get a list of them which I display as tabs.

And this is how I load the Livewire component in order to get the current ID of the "parent"

```
<livewire:catalog-corpuri
                        id="{{ Route::current()->parameter('id') }}"
                        :wire:key="$corpuri->id"
                />
```

Now what I want to do is when I click on one of these tabs to actually either reload the entire component with the content corresponding to that tab or somehow load a new component with the content for that tab.

**What I tried so far**

I created a `public bool $show = false` and in the view, `onclick` set the `$show` to true. Problem was that no matter on which of the tabs I was clicking it will only be showing the last `$corp` added.

 I was also thinking in using Alpine for this but the Livewire way seems more appealing since the content of each `$corp` will be fetched from the server on click. 

So what I want to achieve is something similar with the Livewire pagination... on click, reload the component with the content corresponding to that `$corp`. 

I don't know where to start. Any advice will be much appreciated.

Thanks in advance.

![Screenshot_35](https://user-images.githubusercontent.com/35864222/117866152-7a84cd80-b29f-11eb-9fc2-4a652fecb1cb.png)

**Comment by ju5t** [comment]
You could do add the id's of the model to the tabs, assuming they are all the same. When you click on one of the tabs you can do:

```
wire:click="$set('selected', 7121)"
```

Or you could have it trigger a method that sets `selected` to the model of the actual item. For example:

```
wire:click="setSelected(7121)"
```

Where `setSelected` would be something like:

```php
function setSelected(int $id)
{
    $this->selected = Model::find($id);
}
```

You can then use the `$selected` variable in your template.

**Reply by mvpopuk** [reply]
Thanks for your response... I got the "selecting" part done. I mean I can see the id's of selected items... now, the next issue is how to display the content of the selected id... for now, I only have {{ $corp->name }} to see if is working. 

Here's the code:

```
<div x-on:click.prevent="tab='{{ $selected }}'" 
        wire:click="$set('selected', {{ $corp->id }})" 
        value="{{ $selected }}" 
        class="flex-col cursor-pointer rounded hover:bg-blue-100">
```

```
@if($selected) 
<div x-show="tab == '{{ $selected }}'" 
        x-cloak 
       class="flex" 
       wire:model="selected" 
       id="{{ $selected }}" 
       value="{{ $selected }}">
                 
<div class="mr-5" >
     {{ $corp->denumire }}
</div>

</div>
```

```
 public function setSelected($id)
    {
        // $this->selected = Corpuri::whereKey($this->selected)->pluck('id');
        $this->selected = Corpuri::find($id);
    }
```

For some reason, if I click on any of the tabs, It is showing me the content for the last tab and if I click on another tab, I can see in the Livewire chrome dev extension that It gets the correct id selected but the content doesn't change.

Thanks in advance.

**Comment by mvpopuk** [comment]
Ok, so I mannaged to make this work (sort of). The problem now is that the output of `$selected` is something like this `[{"id":2,"denumire":"Dulap 2U","prod_id":"1","created_at":null,"updated_at":null}] ` and I want to have this as an object and be able to access it like this: `$object->name`

**Comment by mvpopuk** [comment, marked as answer]
Ok I sorted out and was simpler than I imagined.

Here's the method:
![Screenshot from 2021-05-24 01-06-49](https://user-images.githubusercontent.com/35864222/119278571-c7b55780-bc2e-11eb-87d7-1fa023fbc2be.png)
I pass the `$id` to the method from blade like this:
![Screenshot from 2021-05-24 01-07-39](https://user-images.githubusercontent.com/35864222/119278577-d6037380-bc2e-11eb-9634-3511f972c72a.png)
And then if `$selected` has value I can use the object:
![Screenshot from 2021-05-24 01-08-36](https://user-images.githubusercontent.com/35864222/119278585-e3206280-bc2e-11eb-9e9a-390f3d55245c.png)

**Reply by ju5t** [reply]
That's pretty much the exact same thing as I answered a while ago 😅

**Reply by mvpopuk** [reply]
Yes, thanks it's just for some reason I was not checking if `$selected` has value in blade so on `render()` was trying to initialize `$selected->something` and will throw an error since the variable `$selected` was empty at that point.
